src/NCPadWidget.cc

```diff
@@ -323,8 +323,32 @@ void NCPadWidget::wCreate( const wrect & newrect )
 				    padrect.Sze.H, padrect.Sze.W,
 				    padrect.Pos.L, padrect.Pos.C,
 				    'r' );
-	hsb = new NCScrollbar( *this, *win, wpos( win->maxy(), 1 ), win->width() - 2, NCScrollbar::HORZ );
-	vsb = new NCScrollbar( *this, *win, wpos( 1, win->maxx() ), win->height() - 2, NCScrollbar::VERT );
+
+        // Scrollbar has to be at least one character large otherwise ncurses window
+        // for the scrollbar will fail to create and we end with an exception which
+        // crashes whole UI consequently.
+        //
+        // scrollbar size is lowered by -2 bcs there is an overhead for frames etc.
+        if(win->width() - 2 > 0)
+        {
+            hsb = new NCScrollbar( *this, *win, wpos( win->maxy(), 1 ), win->width() - 2, NCScrollbar::HORZ );
+        }
+        else
+        {
+            // no space no scrollbar, scrolling still works using arrows
+            hsb = 0;
+        }
+
+        if(win->height() - 2 > 0)
+        {
+            // we have enough space for vertical scrollbar
+            vsb = new NCScrollbar( *this, *win, wpos( 1, win->maxx() ), win->height() - 2, NCScrollbar::VERT );
```

Comment:
The number `2` is used quite often here, I'd prefer using some constant or macro for that. It would be easier to change it if needed...

Reply:
constantized